src/test/ui/consts/assert-type-intrinsics.stderr

```diff
@@ -1,9 +1,17 @@
 error: any use of this value will cause an error
-  --> $DIR/assert-type-intrinsics.rs:14:9
+  --> $SRC_DIR/core/src/mem/maybe_uninit.rs:LL:COL
+   |
+LL |               intrinsics::assert_inhabited::<T>();
+   |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   |               |
+   |               aborted execution: attempted to instantiate uninhabited type `!`
+   |               inside `MaybeUninit::<!>::assume_init` at $SRC_DIR/core/src/mem/maybe_uninit.rs:LL:COL
+   |               inside `_BAD1` at $DIR/assert-type-intrinsics.rs:14:9
+   |
+  ::: $DIR/assert-type-intrinsics.rs:13:5
    |
 LL | /     const _BAD1: () = unsafe {
 LL | |         MaybeUninit::<!>::uninit().assume_init();
-   | |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ aborted execution: attempted to instantiate uninhabited type `!`
 LL | |     };
```

Comment:
So you think this is better than before?

I honestly think hiding these frames here makes the error look better, and was wondering if it would make sense to do this in Miri, too (unless MIRI_BACKTRACE=full is set), because the user rarely wants to see stdlib frames.

Reply:
> Even with std sources, I feel like pointing at assume_init is better than pointing at its insides.

I very much agree. Otherwise the error points users at code that they've never even seen, when they make a simple error. That's exactly why we're using `#[track_caller]` in so many places in `std`, to make sure a useful location is reported.

Reply:
> we don't prune regular backtraces either...

This isn't really a regular backtrace, though. This is part of a compiler diagonstic, where we are (and can be) much friendlier.

Reply:
Okay, I took out the last commit.

@oli-obk is that fine for you?